src/pickPackageFromMeta.ts

```diff
@@ -77,6 +77,18 @@ function pickVersionByVersionRange (
   if (versionRange === '*' || semver.satisfies(latest, versionRange, true)) {
     return latest
   }
-  const maxVersion = semver.maxSatisfying(versions || Object.keys(meta.versions), versionRange, true)
+  versions = versions || Object.keys(meta.versions)
+
+  const maxVersion = semver.maxSatisfying(versions, versionRange, true)
```

Comment:
Why you don't filter out deprecated versions first? You can check if `versions.length > 1` and filter only on that condition, then call `semver.maxSatisfying`.

Reply:
My assumption is that in >90% of cases the version will be not deprecated. So from performance perspective it is better to filter the version only when the regular selection got a deprecated version

Reply:
👍